### PR TITLE
feat: add health check route

### PR DESCRIPTION
**Pull request by iRohitGaur**
## Issue Ticket Number
#85 

## Description

- Added Health check route
- Added unit test for health check

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
Screenshot 2024-01-27 at 12 50 46

## Test Coverage
Screenshot 2024-01-27 at 12 49 53

### CODE REVIEW

File: skill-tree/src/main/java/com/RDS/skilltree/Health/HealthCheckController.java
Comment (debanjanc01) [inline skill-tree/src/main/java/com/RDS/skilltree/Health/HealthCheckController.java:24]: Just a thought, if we are using actuator, why not use the actuator `/health` endpoint? Do we need a separate health controller in that case?

Comment (iRohitGaur) [reply skill-tree/src/main/java/com/RDS/skilltree/Health/HealthCheckController.java:24]: Hi @debanjanc01 , 

Thank you for your review and the insightful questions.

Actuator's /health endpoint is comprehensive for general health status, our custom implementation lets us tailor the health information and format it according to our specific requirements.

Spring Boot Actuator's /health endpoint can provide uptime information, but this is not included by default. The /health endpoint typically shows basic health information about the application, like database connectivity, disk space, and other custom health indicators. 

`/actuator/health`

> {"status":"UP"}

`/actuator/health/db`

> {"status":"UP","details":{"database":"MySQL","validationQuery":"isValid()"}}

`/actuator/health/diskSpace`

> {"status":"UP","details":{"total":499963174912,"free":99603329024,"threshold":10485760,"path":"/~/rds-skill-tree-backend/.","exists":true}}


On the other hand, the /metrics endpoint `/actuator/metrics/process.uptime` will give this result by default:

> {"name":"process.uptime","description":"The uptime of the Java virtual machine","baseUnit":"seconds","measurements":[{"statistic":"VALUE","value":22.649}],"availableTags":[]}

Hence, our custom approach will allow for more specific control over the health data being exposed, which includes the uptime. It will also help in formatting the output, and, in future, include additional details specific to the application's needs that are not provided by Actuator's standard endpoints.

Comment (debanjanc01) [reply skill-tree/src/main/java/com/RDS/skilltree/Health/HealthCheckController.java:24]: Hi @iRohitGaur, in that case I'm assuming that the actuator has been added for current/future functionalities, while keeping the custom health controller.

With actuator in place, since the `process.uptime` metric gives the value for uptime, I'm trying to understand the reason for registering a `custom.uptime` gauge. Are we looking for a different value?

Comment (iRohitGaur) [reply skill-tree/src/main/java/com/RDS/skilltree/Health/HealthCheckController.java:24]: Hi @debanjanc01 ,

Thank you for pointing this out. I have updated the code and removed the unnecessary `custom.uptime` guage.

**Review by prakashchoudhary07 — APPROVED** [review]
LGTM, We can merge it after one more approval

**Review by bhtibrewal — APPROVED** [review]

**Review by bhtibrewal — COMMENTED** [review]
Okay to merge after @debanjanc01's approval

**Comment by debanjanc01** [comment]
> Okay to merge after @debanjanc01's approval

**Comment by debanjanc01** [comment]
@iRohitGaur closed it by mistake. Have reopened it again.

**Pull request merged**
